modules/commons/src/main/java/org/apache/synapse/commons/json/jsonprocessor/validators/ArrayValidator.java

```diff
@@ -0,0 +1,298 @@
+    private static final String MAX_ITEMS = "maxItems";
+    private static final String ITEMS = "items";
+    private static final String UNIQUE_ITEMS = "uniqueItems";
+    private static final String ADDITIONAL_ITEMS = "additionalItems";
+
+    /**
+     * This method will validates an input array according to a given schema.
+     *
+     * @param input  input array as a Map.
+     * @param schema JSON schema as an object.
+     * @return Validated JSON array.
+     * @throws ValidatorException Exception occurs in validation process.
+     * @throws ParserException    Exception occurs in data type parsing.
+     */
+    public static JsonArray validateArray(Map.Entry<String, JsonElement> input, JsonObject
+            schema) throws ValidatorException, ParserException {
+
+        JsonParser parser;
+        int minItems = -1;
+        int maxItems = -1;
+        boolean uniqueItems = false;
+        boolean notAllowAdditional = false;
+        // parsing the properties related to arrays from the schema, if they exists.
+        if (schema.has(UNIQUE_ITEMS)) {
+            String uniqueItemsString = schema.get(UNIQUE_ITEMS).getAsString().replaceAll(ValidatorConstants.REGEX, "");
+            if (!uniqueItemsString.isEmpty()) {
+                uniqueItems = DataTypeConverter.convertToBoolean(uniqueItemsString);
+            }
+        }
+        if (schema.has(MIN_ITEMS)) {
+            String minItemsString = schema.get(MIN_ITEMS).getAsString().replaceAll(ValidatorConstants.REGEX, "");
+            if (!minItemsString.isEmpty()) {
+                minItems = DataTypeConverter.convertToInt(minItemsString);
+                if (minItems < 0) {
+                    throw new ValidatorException("Invalid minItems constraint in the schema");
+                }
+            }
+        }
+        if (schema.has(MAX_ITEMS)) {
+            String maxItemsString = schema.get(MAX_ITEMS).getAsString().replaceAll(ValidatorConstants.REGEX, "");
+            if (!maxItemsString.isEmpty()) {
+                maxItems = DataTypeConverter.convertToInt(maxItemsString);
+                if (maxItems < 0) {
+                    throw new ValidatorException("Invalid maxItems constraint in the schema");
+                }
+            }
+        }
+
+        // parsing additionalItems
+        // We are wrapping the additionalItems schema inside a json array schema object and calling this same method
+        // again
+        JsonObject additionalItemsSchema = null;
+        if (schema.has(ADDITIONAL_ITEMS)) {
+            JsonElement tempElement = schema.get(ADDITIONAL_ITEMS);
+            if (tempElement.isJsonPrimitive() && !tempElement.getAsBoolean()) {
+                notAllowAdditional = true;
+            } else if (tempElement.isJsonObject() && !tempElement.getAsJsonObject().entrySet().isEmpty()) {
+                String jsonString = "{\"type\": \"array\",\"items\": " + schema.get(ADDITIONAL_ITEMS).toString() + "}";
```

Comment:
Let's use StringBuffer instead of string concatenation

Reply:
Fixed